core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -554,25 +549,31 @@ private object RTS {
                     val io = curIo.asInstanceOf[IO.Supervise[E, Any]]
 
                     curIo = enterSupervision *>
-                      io.value.ensuring(exitSupervision(io.error))
+                      io.value.ensuring(exitSupervision)
 
                   case IO.Tags.Terminate =>
                     val io = curIo.asInstanceOf[IO.Terminate[E, Any]]
 
-                    val cause = io.cause
+                    val causes = io.causes
 
                     val finalizer = interruptStack
 
                     if (finalizer eq null) {
                       // No finalizers, simply produce error:
                       curIo = null
-                      result = ExitResult.Terminated(cause)
+                      val allCauses = status.get match {
```

Comment:
Ditto for this, I think we push this to `causes` method on the status so the code is just `io.causes ++ status.get.causes`.